### DISCUSSION TITLE
More maintainers and reviewers

### DISCUSSION THREAD

**Discussion by elidrissidev** [OpenMage/magento-lts#1730] [General]
Hello everyone,

In my company, we are currently in the process of migrating to OM with PHP 7.4 from Magento CE with PHP 5.6. Because of how much time and pain this project saves us, I lately decided to start contributing to this project and giving back to the community. I'm very happy this exists.
The problem that I'm noticing is, there is a lot of pending Issues and Pull Requests, some with no reviews, some with partial reviews... etc.
I understand that the maintainers have other things in life, but my suggestion is to get some other *trusted* people that can review and approve/reject changes.

✌️

**Comment by Flyingmana** [comment]
In general, every contributor is allowed to provide Reviews to Pull Requests. It may not provide the "green checkmark", but the maintainers still take them into evaluation for merging. 
And some of us prioritise PRs which already have one or two Reviews.
In the past we did also promote some of the regular Reviewers, so they can also contribute to the green checkmark, and merge themselve when it is reached.

Besides that, we try to not downright reject changes, but try to find ways on how the desired goal can be reached. Except its a major Security or BC Break.

**Reply by fballiano** [reply]
@Flyingmana what you say it totally right but it does not happen, too many PRs have no reviews or too little reviews so they stay in limbo forever, it does not make any sense, it's blocking the development and making people lose interest.

I brought this up already some time ago and I still think it's the main problem of this project.

OM has some AMAZING developers participating, but they can't do everything by themselves (for obvious reasons) but everybody else is sadly kept outside.

**Reply by Flyingmana** [reply]
Oh, do you have an example of someone kept outside? Thats absolutely something which we want to avoid. Everybody should be able to participate as much as they want.

**Reply by fballiano** [reply]
we're kept outside when we can't decide anything, if we open PRs that stay there forever cause there's not enough people to review them then the "regulation" is wrong and needs to be changed.

**Reply by Flyingmana** [reply]
If you think the regulation should change, the community can change them via the RFC process.
If the majority of the Community thinks, fewer Reviewers should be required for a merge, then its their fair right to get this change.

The RFC process grants that all point of views get heard, and that there is enough time for everyone to comment and vote on it.

So basically, you can decide everything, but you need to put work into it, and follow the process.

**Reply by fballiano** [reply]
I think a lot of people are putting the work in... it's pretty clear from the amount of PRs

**Reply by Flyingmana** [reply]
you misunderstood me. What Iam saying is, to decide something, to change a decision, to change the "regulation", you need to put work into creating an RFC.

**Reply by addison74** [reply]
@elidrissidev: I agree that a cleanliness is required in the two mentioned sections. I recently asked it for a couple of times. I personally cleaned up everything I posted over time. I kept only OM issues in the Issues section and posted in the new Discussions section all the ideas I would like to see implemented.  

Although I asked questions or had ideas in certain posts, those who were involved in the discussions over time did not respond. Obviously, those who are in charge of maintaining this project must also be in charge of evaluating these Issues/PR's that have been lying there for a long time. 

OpenMage is a very cool project and I've noticed lately that people have come back that I haven't seen in a while. OpenMage really is a functional mature platform and we have a huge legacy left by the late Magento 1.

**Reply by elidrissidev** [reply]
@Flyingmana Two reviewers are fair enough in my opinion, nothing against that. But adding more volunteers as maintainers, I think, would help a lot.

**Reply by fballiano** [reply]
@elidrissidev exactly, more people should be allowed to review/merge, not everybody, but somebody more

**Reply by Flyingmana** [reply]
again, everybody is allowed to review.

**Reply by fballiano** [reply]
again, is this a joke? are there 300 PRs why?? because nobody merges them or reject them! because there's the need for a couple (at least) more people with privileges on this repo!

**Reply by Flyingmana** [reply]
@fballiano I think what you want is not more people who are allowed to review/merge, but more people with a free pass to close/decline issues/PRs, which is a complete different topic. It is different, because while you can technically close them already with a very low rank, community wise you are very fast with offending people so that even as a maintainer you cant just close an issue/PR so easily.
Its one of the main differences between us and Magento2. Its the difference why some people contribute to us, who already gave up to contribute to Magento2. Its the reason why I already gave up to contribute to several OpenSource projects. 

But still, its something which can be changed with an RFC. Then its a community decided rule and I will not try to block anything in this regard anymore.

**Reply by addison74** [reply]
There are many posts where someone asks for the author's opinion but he does not respond. Some rules should be established like we see in other active projects. If the author does not provide feedback within a certain period, the post will be closed. It can be opened later. There are others where although he was answered, although he is asked to close the post, the author does not want to do it. 

We need to focus on active issues and PR's. Magento 1 has been buried for more than a year but that doesn't mean that OpenMage has the same fate. When you see activity it is like a magnet that attracts many interested people. I remain of the opinion that many do not know that there is an alternative after Magento 1. That's why I said that a person with notoriety should talk to extension developers and get them to keep the products for sale. I know most of them wanted to get some not inconsiderable money by migrating to Magento 2, but things didn't always go well. I know people who regret changing the platform, they hate the interface and most importantly the new costs.

**Reply by fballiano** [reply]
I think nobody offended nobody refusing a PR until now and I don't see why that would happen with a better moderation team.
You know what offends me? when somebody takes hours to create a PR and nobody cares.
That is how project die, not the other way around.

Does somebody really think that the authors of these 180 opened PRs are happy that they worked for nothing? I'm not happy that I worked 6 hours fixing the ugly logos mess just to hear "nah I like the ugly one better".

I try to give feedback on as many PR/issues as possible but while active people try to do that, when they contribute, nobody gives feedback to them. This system is punishing the active people trying not to offend the inactive ones.

The system should give something to active members, you have 10 approved PRs? ok now you've some more privilege. Here it's the other way around.

**Reply by Flyingmana** [reply]
there were around 4 cases I know of (where I did mitigation afterwards), plus 2 issues of myself which were closed  because they were not understandable by the one closing, where I felt very offended.

**Reply by fballiano** [reply]
whatever, more people are trying to tell you the same thing since long time and, I don't understand why, you don't want to listen. It feels exactly like Magento2.

just a couple of PRs laying around DESPITE reviews:
- https://github.com/OpenMage/magento-lts/pull/1628
- https://github.com/OpenMage/magento-lts/pull/1637
THIS offends me, to see good works thrown to garbage!

**Reply by addison74** [reply]
Case Study. splide.js a very cool javascript vanilla that came to offer many features to already suffocated projects like Owl Carousel. I was among the first to test it and I had a few observations, then I contributed to its promotion. As it was natural, interest exploded exponentially, many people either reported problems or came up with improvements. The problem is that the person who maintains the project is no longer as active as at the beginning when in no time he responded to requests. The inevitable happened and the first signs of frustration appeared. PR's that have been lying for a long time, real problems that have not been answered. It is natural that in such situations the initial energy dissipates and people cause other solutions. Let's remember Turpentine project where adding Varnish support to Magento 1 was a real improvement in dealing even with small DDoS attacks.

**Reply by fballiano** [reply]
But also you're saying "closing a PRs offends people" so "let's just simply not answer and leave it there forever"? Like that would be a better option?

**Reply by Flyingmana** [reply]
> whatever, more people are trying to tell you the same thing since long time

mostly the same few, which is why I dont think its the majority. 
 But thats why there is the RFC process, to determine if its the majority or not. But everybody just focuses on trying to convince me

**Reply by fballiano** [reply]
like there are 100 people constantly active...

**Reply by elidrissidev** [reply]
@Flyingmana Can you please elaborate on how to create an RFC?

**Reply by Flyingmana** [reply]
@elidrissidev similar, its by creating a document in https://github.com/OpenMage/rfcs
The Readme describes the current process

**Reply by fballiano** [reply]
@elidrissidev I'm creating a RFC here https://github.com/fballiano/rfcs/blob/main/draft/0000-change-pr-merging-rules.md

**Reply by fballiano** [reply]
@elidrissidev feel free to share your ideas and let's create this PR (and later one to define rules to close the abandoned Prs)

**Reply by elidrissidev** [reply]
@fballiano Alright will check it out in a few hours. Thanks.

**Reply by addison74** [reply]
Reading the current RFC for accepting PR's I can say it is not wrong either, but too many PR's have remained open for a long time. Here is the big question, when a PRs is good, complete or unfinished? I do not want any haste in their approval because at least in the last half year I woke up that either a file was accidentally deleted (color picker), the installer did not work and many other issues that had the approval of some people with the experience. 

Those who approve must check that PR not just click on a link because they think it was okay to read the code. Those of us who have time should take all 180 PR's in hand and express our opinion if they remain open, closed, or being integrated into OM. Here is the big issue the reaction time. And some PR's are even complex and require attention, but what happens to them must be established once and for all in the next months. I also look at the activity of the authors on GitHub when I try to contact them in their PR's. In other words, good PR's can be taken over by other authors and carried on.

**Reply by fballiano** [reply]
@ADDISON74 that is why I'm trying to define a RFC to change the rule. bugs will happen, OM will never have the power of a corporation, that doesn't mean we've never change anything. the v19 is the stable one, v20 if you want improvements. something will break, everybody is careful, everybody here have multiple OM projects in production and nobody wants to see them broken.

Anyway I think a PR is complete when is approved.

**Reply by fballiano** [reply]
following the RFC rules I had to change the proposal address to https://github.com/fballiano/rfcs/blob/main/draft/0000-change-pr-merging-rules.md I've also drastically updated it (introducing a minimum number of merged PR to be part of the approving users group), any contribution is welcome to define something that works for everybody and solved the problem.

**Reply by addison74** [reply]
A level of priority of applying a PR should also be considered. Some need to be applied urgently and others may wait. There were situations this year when new versions were launched and issues were reported in the same day but they were not solved immediately, some being fixed even after a week or more. I am talking about issues that affect the current functioning of OM and that cannot be postponed.

**Comment by kiatng** [comment]
@fballiano Thank you so much for your RFC. Currently, OM has only 1 merge criteria: 2 green checks. The grey checks doesn't count, even though @Flyingmana says the greys add weigh to the PRs for it to get merged. Your RFC proposes to add an additional merge criteria, 1 green + 1 grey checks. **I support this**. However, I would prefer to have 1 green + 2 greys. This would delay the merge a little bit but grey reviewers are more active anyway in OM, so it shouldn't cause too long a delay. So, if a PR has 1 green + 2 greys, then it meets the merge criteria and should be merged. Does that make sense? Further, I would propose that a PR with 3 greys without anyone objecting should be considered as a third criteria.

**Reply by fballiano** [reply]
@kiatng thanks for your message! yesterday night I changed my idea a bit and the new RFC (which was published before your message for sure.. that's kinda strange) was about allowing "green checks" to contributors that have at least 5 merged PRs (to avoid random users give grey checks and pollute the system) although I'm not sure that this is actually possible with github features. So my latest idea would be, for a PR to be approved, to have "2 maintainers with positive review" or "1 maintainer + 1 of this new group of users".

I totally agree with your 1green + 2grey (it was in my "alternative solutions" sections) because "1green + 1grey" it's too loose, my question (as of yesterday night) is: is every grey equal or should we have some other threshold (that's why I was thinking a minimum number or merged PRs)?

**Reply by Flyingmana** [reply]
>  was about allowing "green checks" to contributors that have at least 5 merged PRs (to avoid random users give grey checks and pollute the system) although I'm not sure that this is actually possible with github features.

should be possible, basically everything available via githubs API can be used to determine this.
In worst case we could fall back to using a simple textfile list, (which can also get updated automatic)

**Reply by fballiano** [reply]
@Flyingmana I've created a quick php+curl script and to get the list of the 28 contributors (some of them are already maintainers) with more than 5 "contributions" (which should be commits), it uses my personal access token so I can't directly publish it but I'll update the RFC with it.

This is the API endpoint: https://api.github.com/repos/OpenMage/magento-lts/contributors

**Reply by addison74** [reply]
Filtering Issues section I discovered there are 16 **Questions** and 36 **New Feature** marked by labels. There are more than that, some issues are opened for a long time event the authors did not reply or some of them are fixed. I suggest the maintainers to ask the authors for those marked as "New Feature" to be posted in Discussions section. In this way we can continue evaluation and discussions from there.

**Reply by Flyingmana** [reply]
@ADDISON74  thats not thought very far. Why should a new Feature be posted to Discussions, if it is a clear ToDo which needs someone to take over? example the first in the list: https://github.com/OpenMage/magento-lts/issues/1726

**Reply by addison74** [reply]
@Flyingmana - we can't mix apples with pears. The Issue section should be dedicated only to issues identified in the OpenMage code. Anyone who has a real problem can report it and it will be analyzed and solved together with the others. A new feature in OM means discussions that can be resolved elsewhere. 

I moved all my personal suggestions in the Discussions section because there are new features not issues in the code. If anyone is interested in these proposals they can make PR's. What I'm trying to do is keeping a strict section for real issues in the OM code. I do not hide the fact that from time to time I check this Issue section and it bothers me that I do not always find real confirmation problems. I often waste my time with particular, unconfirmed situations and this is not good at all.

**Reply by addison74** [reply]
If those who maintain this project want to get involved in marking those posts that have really died they should let us know. We can establish as a rule that when 2-3 active people here mark a post that lacks interest or reaction from the author to be closed. Thus one like you who maintains the project can no longer be accused of committing a solitary action of closing.

**Reply by fballiano** [reply]
Hi everybody, I've rewritten my RFC completely: https://github.com/fballiano/rfcs/blob/main/draft/0000-change-pr-merging-rules.md but before submitting it I'd like to post it here to get some feedbacks. I hope it's a better version compared to the first one!

Thanks! :-)

**Reply by addison74** [reply]
From my perspective what you have proposed is primarily to be appreciated because I have found that fewer and fewer people are showing interest in this project. Secondly we have a big problem both in the reported issues and in the PR. Their number is huge. We seem to have been stepping on the spot for some time. I agree with what you proposed and thank you for your work.

**Reply by kiatng** [reply]
@fballiano Awesome! I support your RFC. Thank you.

**Reply by midlan** [reply]
I don't think openmage need to change the merge rules. I think more users should be privileged as maintainers. I belive there is many companies running openmage. One or two people from that companies should attend as maintainer in openmage and push forward their priorities. Otherwise they must be waiting to other maintainers that are doing their job for free (see my post about sponsorship below), which take a lot of time, because people tend to do first the things that thay are paid for, later the free ones :)

**Reply by fballiano** [reply]
I think that there should be a "rule" on how to be(come) a maintainer, is there now?

**Reply by midlan** [reply]
I think there isn't. I became maintainer just after some discussion with flyingmana.

**Reply by mark-netalico** [reply]
Maybe there should be a submission form to apply to become a maintainer and X number of current maintainers need to approve it?

**Comment by midlan** [comment]
For that reason, I think openmage should start some sponsorship program. Companies that love this project would be paying for it. Even our company would be paying for it. Then the maintainers would be motivated much more to push things forward :)

**Reply by Dirtyhair1** [reply]
What do you propose with this? How would it work in reality? Checking Builtwith there are around 19500 live stores using OpenMage, https://trends.builtwith.com/shop/OpenMage, I also agree that some of those store owners might be interested in contributing.

**Reply by Flyingmana** [reply]
For starting a Sponsorship program, someone would need to setup a legal entity for this. I know @mark-netalico was working on something in context of a bugbounty programm, but I did not see any progress there.
I also know there were at least a few who offered to provide money as sponsorship.
If anyone know of someone, who would be willing to take over the involved bureaucracy to do this, happy to help there.
Besides this, to speak only for me, you can sponsor me directly via Githubs Sponsorhip programm already.

**Reply by mark-netalico** [reply]
I was trying to refine the scope of the Hacker One bug bounty program so we weren't getting so much spam. I think it's pretty close, but I've been more aggressive about closing things related to the OpenMage *server* and not any of the code. To be competitive with other similar projects it would be fairly expensive: https://mageone.atlassian.net/wiki/spaces/MOBB/pages/815792139/Rewards. Since we're a more not-for-profit project we might could offer a lower bounty.

In terms of other sponsorships payout, I'm open to ideas for the criteria/amount.

So basically I think we need to formalize how/what we payout for what and then we can move forward.

After we have that decided, I'm happy to open up the legal entity and go ahead and contribute a significant sum to start it off. I'd want to keep an "open ledger" of money-in/money-out. If people need to be anonymous we can redact some people from the open ledger, but internally we'll still need to record that and other members of the OpenMage team could see.

**Reply by Dirtyhair1** [reply]
@Flyingmana I understand what you are saying, but is setting up a legal entity maybe a bit complicated for an initial start? I mean, if it gets momentum, yes, it makes sense but at the very beginning, couldn't there be a simple account created on Transferwise, WorldPay or even Paypal and deposits and payments can be made from there? There could be transparency around it? Just trying to keep things simple to start with....

**Reply by Dirtyhair1** [reply]
> So basically I think we need to formalize how/what we payout for what and then we can move forward.
> 
> After we have that decided, I'm happy to open up the legal entity and go ahead and contribute a significant sum to start it off. I'd want to keep an "open ledger" of money-in/money-out. If people need to be anonymous we can redact some people from the open ledger, but internally we'll still need to record that and other members of the OpenMage team could see.

I am one of those merchants. I am happy to contribute to this project, but also I would prefer to do it anonymously if possible. I am happy to be on the private record but would prefer my name/business redacted from public viewing. I think most merchants (not all though) would want this.

But this seems to be 2 different ideas here:

1) money for bug bounty which is what we are talking about.

2) @midlan mentioned a sponsorship program mainly so things like PR's can get moved along. That's kinda different to a bug bounty program.

**Reply by Flyingmana** [reply]
@Dirtyhair1 for many of us, there would be taxes involved and possibly additional fees by taking the money into an account which is assigned to us personally.
Depending on Country and circumstances, this could lead to around 50% of loss.
A legal entity can with a bit of paperwork reduce this closely to 0%., or at least a few static cost.

PayPal would be a terrible choice for this, as they like to close these kind of accounts, or block the money flow for months.
And worse, if they do this, they include also all the personal payPal accounts connected to the people.
Also having more then one PayPal account is a ToS Violation, so opening one without a legal entity will directly kill all private (and possibly other Business) PayPal accounts connected to the people managing them. (FOR EVER!!!)
Also you still need a real bank account as soon as you get over I think 3,000€, else the funds are frozen.

**Reply by Dirtyhair1** [reply]
@Flyingmana yes I understand what you mean, I just mentioned Paypal but now understand it's not a great idea. So what are you thinking here? I also maintain my comment above:

"But this seems to be 2 different ideas here:

    money for bug bounty which is what we are talking about.

    @midlan mentioned a sponsorship program mainly so things like PR's can get moved along. That's kinda different to a bug bounty program."

Which are you thinking about/suggesting? Or both?

**Reply by Dirtyhair1** [reply]
@Flyingmana ok I have been thinking about this a bit more and now I understand what you mean about taxes etc. I agree an entity would be best, also better for the long term. There could be 2 separate accounts. Those who want to sponsor for bug bounty could donate into one account, and those who want to sponsor for other purposes could donate into another account, that’s my first thought at least.

**Reply by JonLaliberte** [reply]
I wonder if something like the route Home Assistant went could work for OpenMage? Basically they started a for-profit that extends the project ([Nabu Casa](https://www.nabucasa.com/about/)). With the funds raised from that feature, they have brought on developers full time to enhance the open source project and allow them to concentrate on it full time.

**Comment by f1-outsourcing** [comment]
It is to me also not entirely clear what is being allowed into the master. I think a decision should be made whether or not magento-lts is moving forward or just receives security updates.

Eg. the default template requires an update
https://github.com/OpenMage/magento-lts/issues/1847#issuecomment-937606408

Eg. apcu is this now added? 
I am not sure how long this project exists. But to me it is weird that nobody, until now noticed this is not working.
https://github.com/OpenMage/magento-lts/discussions/1795

Eg. allowing different email templates to be stored within theme location.
https://github.com/OpenMage/magento-lts/issues/1773

Eg. allow recurring payments for bank transfers
I am currently trying to get recurring bank transfers to work. I am not entirely sure if I understand the recurring profiles correctly, and why this is not enabled by default for bank transfers. But to me this looks like nice to have.
https://github.com/OpenMage/magento-lts/discussions/1940
https://github.com/OpenMage/magento-lts/commit/141fe9b043390def90d652bc10b3737bc3fbfb24

**Reply by Flyingmana** [reply]
for apcu, its not used much, as only few have it actually installed/enabled. Usually people use redis for the cache

for recurring payments, the feature was I think never fully finished/working by itself, but I also never did need to use them.

For the main question, moving forward, but in a way that is not putting to much work on existing installations.

**Reply by fballiano** [reply]
I alreay knew that recurring payments were always in beta stage and never fully completed.

**Reply by f1-outsourcing** [reply]
Although I am not really familiar with the magento structure, I have 'I think' something working with recurring payments for the bank transfers. Just some proof of concept without any exception handling etc.

**Reply by f1-outsourcing** [reply]
I see the magento-lts as the entry level stores for people that do not want(need) to install magento 2 + elastic search + ... . any other crap (yet). It is good for small solutions, and does not require a large footprint.

In that line of reasoning, one should offer the apcu it is just a quick win, without having to need to install some 3rd party solution.

The same is for this issue with these email templates being grabbed from wrong location. The whole idea of a smooth and fast installation is to unpack the magento-lts source, dump your theme and other modules in it, and preferably run  a group of n98-magerun.phar commands and you are up and running.

Maybe it is good to send some questionnaire to see what the magento-lts audience is like.

**Reply by addison74** [reply]
I have been with OpenMage for almost 2 years and I can say that this project is based on the contribution of those who still use the Magento 1 platform. It is a voluntary, unpaid contribution. At this time 71 people are receiving notifications about changes made to OpenMage. There is a list of issues, one for PR's, and another one for discussions. Although there are many stores that still use M1, those who manage them do not participate here. Otherwise we would see an increased interest.

APCu is the successor to APC, in the past I used it when PHP was at version 5, but starting with version 7 the Zend OpCache extension is the one I use. The same thing happened using Redis, all the sessions are stores in the system file. In addition, I have a dedicated server like I didn't have in the past with NVMe, RAM, a lot of processor cores. I could use Varnish cache but for the moment all stores I managed are loading the pages fast. I can deal easy with 200 simultaneous connections and for DDoS I am using a hardware solution for which I pay monthly subscription.

I doubt you'll find someone to bring APCu extension up to date. With a few exceptions, most developers have taken their extensions out of sale and services for M1. I don't know how long OpenMage will last, but it is the result of the contribution of each of us, and I'm glad there are still people interested.

I evaluated a lot of shopping carts like Woo, Shopware, OpenCart, Prestashop. I'm ready to migrate to Magento 2 when I'm forced to. We will see what issues the new PHP versions will involve, here is the key to the survival of M1-based stores.

**Reply by f1-outsourcing** [reply]
> I have been with OpenMage for almost 2 years and I can say that this project is based on the contribution of those who still use the Magento 1 platform. It is a voluntary, unpaid contribution. At this time 71 people are receiving notifications about changes made to OpenMage. There is a list of issues, one for PR's, and another one for discussions. Although there are many stores that still use M1, those who manage them do not participate here. Otherwise we would see an increased interest.
> 
> APCu is the successor to APC, in the past I used it when PHP was at version 5, but starting with version 7 the Zend OpCache extension is the one I use. The same thing happened using Redis, all the sessions are stores in the system file. In addition, I have a dedicated server like I didn't have in the past with NVMe, RAM, a lot of processor cores. I could use Varnish cache but for the moment all stores I managed are loading the pages fast. I can deal easy with 200 simultaneous connections and for DDoS I am using a hardware solution for which I pay monthly subscription.
> 
> I doubt you'll find someone to bring APCu extension up to date. With a few exceptions, most developers have taken their extensions out of sale and services for M1. I don't know how long OpenMage will last, but it is the result of the contribution of each of us, and I'm glad there are still people interested.
> 
> I evaluated a lot of shopping carts like Woo, Shopware, OpenCart, Prestashop. I'm ready to migrate to Magento 2 when I'm forced to. We will see what issues the new PHP versions will involve, here is the key to the survival of M1-based stores.

I already showed some example of how to update to apcu (it is sort of working in my environment), I am not eager to submit pull requests because of my inexperience with magento. I think some experienced developer would generate a more efficient result.

https://github.com/OpenMage/magento-lts/pull/1946

https://github.com/OpenMage/magento-lts/pull/1948

**Comment by addison74** [comment]
The advantage of having more maintainers is that a PR can be edited/modified by them once issues are reported. If we look at the number of PR's there are so many in which some of us requested revision or update but we did not receive any response from the authors.

If there are more maintainers then you can increase the number of approvals by a PR needs to be merged. We have to do something quick. We have been talking about PR's for almost 2 years and things are not moving at all. 

We should decide what to do with each PR periodically. I think that if we make a plan we can solve the vast majority in a maximum of 3 months.

**Reply by luigifab** [reply]
I lost the link of @fballiano that list PR, but there are some PR with some "gray approve".

**Reply by addison74** [reply]
@luigifab - check this out https://github.com/fballiano/rfcs/blob/main/draft/0000-change-pr-merging-rules.md. At the bottom of the page there is a link.

**Comment by addison74** [comment]
Another advantage would be that keeping active maintainers can debate closing a resolved reported issue or without feedback. In the Issues section today I saw a few posts that have solutions, the authors are asked to provide feedback or close them, but no reaction. We need to do a little cleaning after a while.

**Comment by fballiano** [comment]
Hi everybody, I've just posted the PR for the RFC here:
https://github.com/OpenMage/rfcs/pull/4

I don't know how many maintainers or the RFCs repo are reading this but for it to be merged there's the need for the approval of 2 maintainers. 🤞🤞🤞

**Comment by f1-outsourcing** [comment]
Is something now changing here for the better or not? I do not even get why this apcu is still not added, why is nothing being done here? If it is not possible to get 2 maintainers to approve something, then switch to 1. 
Are these maintainers by any chance also working for Adobe/Magento 2?

**Reply by luigifab** [reply]
hello dear people, we are not working specially for you, and we have many things to do elsewhere

**Reply by f1-outsourcing** [reply]
Hence my suggestion to switch to 1 maintainer to review. Maybe it is time to ask the question till what level activities may sink in order to stay a maintainer? I would even argue that a passive attitude is pushing away the activities of others. Someone like this @elidrissidev should have an opportunity to contribute.

**Reply by fballiano** [reply]
when my RFC https://github.com/OpenMage/rfcs/pull/4  will be merged the process will change

**Reply by midlan** [reply]
@f1-outsourcing please stop screaming about your apcu. Seems no one is using it at the moment. To test it, it would take few hours of work at least. If you willing to pay for the time spent, I am sure someone will check it for you.

I am maintiner and also my pull requests are there waiting many months, even years, because no one run into problems I am fixing.

So go ahead, create your private fork of magento-lts and merge PRs you need personally. Don't wait for others that they will do it for you for free.

**Reply by f1-outsourcing** [reply]
> @f1-outsourcing please stop screaming about your apcu. Seems no one is using it at the moment. To test it, it would take few hours of work at least. If you willing to pay for the time spent, I am sure someone will check it for you.
> 
Nobody is using the non existent bridge, so we do not need to build the bridge. What is there to test, non-existing functions are replaced by existing. They populate the cache, even if this only a 50% fix, it is better than leaving it 100% not fixed.

> I am maintiner and also my pull requests are there waiting many months, even years, because no one run into problems I am fixing.
> 
As if everything is reported here. You should rephrase that to "because the problems you found were not mentioned by anyone else here." 
The idea of applying fixes is that other people will not encounter the same problems. 
Maybe it is better to have some sort of 'beta' branch then, where people like you can submit to, and others can choose to use this if they like to have these latest commits.
This way you will have also larger testing group and earlier feedback on the applied changes. Every 6 months you can then merge them to the master.

> So go ahead, create your private fork of magento-lts and merge PRs you need personally. Don't wait for others that they will do it for you for free.

Ehhhh, you are asking people here for PR's

**Reply by midlan** [reply]
>Nobody is using the non existent bridge, so we do not need to build the bridge. What is there to test, non-existing functions are replaced by existing. They populate the cache, even if this only a 50% fix, it is better than leaving it 100% not fixed.

Still my statement above applies. Few hours of free work. Plus you didn't even bother with proper filling the pull request template. **You didn't even tell the reason why it should be fixed.** No surprise that there is no interest of such PR.

> Maybe it is better to have some sort of 'beta' branch then, where people **like you** can submit to, and others can choose to use this if they like to have these latest commits.

I'm not the one who is complaining that his PRs are not merged. You are the one. As I told, go and create your fork, name it magento-lts-beta as you suggested, merge all the PRs you want and if people are interested you would have feedback.

> Ehhhh, you are asking people here for PR's

I am not asking for anything. PRs are welcome, not requested. And not guaranteed that they will be merged.

---

You are still complaining that people are not doing what you would like, and you would like it for free of course. Do you remember the last time, someone come to you and said, "hey I need some job to be done by you, for free", and you went and did it? You are doing the basically same thing here. If someone done it, **be respectful**. If not, don't complain or scream _why is nothing being done here_ or _[WTF is with this????](https://github.com/OpenMage/magento-lts/pull/1951#issuecomment-1040458186)_, it's pretty rude.

**Reply by f1-outsourcing** [reply]
People here are asking for PR's, so stop asking for them! Or turn this feature off. If you ask for PR's you are obligated to do something with them, otherwise you are wasting other peoples time. How does that fit in your 'be respectful'.

"Pull requests with unofficial bug fixes and security patches from the community are encouraged and welcome!" Remove this then from the main page or add "but we will merge them after 6 months or most likely never"

I think this attitude of the current maintainers is damaging to this project, without them being aware of it. I suspect there are probably several companies having a business model based on this version that need to apply updates quickly and would be willing to participate more frequently to guarantee some future for their business, this is currently is not possible. Since this is not viable business perspective, they will move away to other solutions. 
I think this community project can grow much bigger especially with the current strategy of magento 2.x, but this growth is being blocked by it's current state. I would also not advice people to use mangeto-lts, if things are more active here and more people start using it and a future is more stable I would definitely change that.

Doing something for free does not mean you can do it unprofessionally or do it how ever you like. Think of how your reasoning would be in:

If your kid gets some sweets for free you also expect these sweets to meet some standards. You do not want these to come from the floor, and have someone say "but they are for free"

Or if your kid has some guardian guiding she/he through traffic, and something bad happens. Then this guardian says "but I am a volunteer, I am doing it for free".

**Reply by luigifab** [reply]
I think we are not obligated to do something with them (like I not obligated to answer issue/pr for my open source repositories).
But in same time, you are right, when nothing happens for them, the future of OpenMage is dark, I calling that: the beginning of the end.

**Reply by fballiano** [reply]
exactly @luigifab, I've been try to say the same thing for months and months, with the current pace the project will die.

let's face it (as we all are M1 developers and we are here cause we love it) nowadays M1/OM it's only used for old projects that didn't want to migrate to anything else, but for new projects everybody goes to shopware or shopify or something else...

**Reply by addison74** [reply]
I don't think it makes sense to justify our actions. I personally appreciate the fact that there are still a handful of people who contribute regularly and that's because they maintain M1-based stores. To be well understood, M1 is still a very high-performance, mature platform, I have come across custom implementations made by absolutely fabulous developer teams. As long as this project continues it is all that matters. I'm sorry I haven't seen posts from important people in a while.

**Reply by midlan** [reply]
I contacted flyingmana with list of potential maintainers. He told me that he know the names and will hopedully make some of them to be new maintainers in week or so.

**Reply by Sekiphp** [reply]
@f1-outsourcing 

> I suspect there are probably several companies having a business model based on this version that need to apply updates quickly and would be willing to participate more frequently to guarantee some future for their business, this is currently is not possible.

If some copanies wants to merge PR quicklier, they can send money to the maintainers. For example @Flyingmana have sponzorship program and If you want you can send him money and he will spent some time with your pull requests. 


I think, that all maintainers here are great experienced M1 developers, but we do not have a time to spend with OM 2 hours a day. Me and @midlan thinks, that we should invite more people to maintainers team.

**Comment by f1-outsourcing** [comment]
:) :) I am glad to see this discussion, you can almost call it progress!!!! :) :)

Is there not someone that can add an extra shop where magento maintainers/developers can create products (from PR's, feature requests or whatever) that a sponsor then can buy.
Than you have a perfect record of in and outgoing transfers, and buyers are anonymous.

But keep in mind such intermediate that facilitates the shop, does not get caught in some dispute when a feature is not implemented as the buyer expected.

**Comment by fballiano** [comment]
Hi everybody, https://github.com/OpenMage/rfcs/blob/main/proposal/0001-change-pr-merging-rules.md was merged in the RFCs repository, but it needs votes to be "accepted".

I've opened an issue https://github.com/OpenMage/rfcs/issues/5 to allow you to vote for the approval of this long awaited change!

Thanks everybody!
